lib/axon/activations.ex

```diff
@@ -351,6 +351,67 @@ defmodule Axon.Activations do
   """
   defn linear(x), do: x
 
+  @doc ~S"""
+  Logsumexp activation.
+
+  $$\log(sum e^x_i)$$
+
+  ## Examples
+
+      iex> Axon.Activations.logsumexp(Nx.tensor([-3.0, -2.0, -1.0, 0.0, 1.0, 2.0, 3.0], names: [:data]))
+      #Nx.Tensor<
+        f32[data: 7]
+        [0.45776283740997314]
+      >
+
+      iex> Axon.Activations.logsumexp(Nx.tensor([[-1.0, -2.0, -3.0], [1.0, 2.0, 3.0]], type: {:bf, 16}, names: [:batch, :data]))
+      #Nx.Tensor<
+         bf16[batch: 2][data: 1]
+        [
+          [0.404296875],
+          [0.404296875]
+        ]
+      >
+
+  """
+  defn log_sumexp(x, opts \\ []) do
+    opts = keyword!(opts, axis: -1)
+    axes = transform(opts[:axis], &List.wrap/1)
+
+    transform({x, axes}, fn {x, axes} ->
+      Enum.each(axes, fn axis ->
+        Nx.Shape.normalize_axis(Nx.shape(x), axis, Nx.names(x))
+      end)
+    end)
```

Comment:
Nx functions will do the normalization of axes internally for you, so you can avoid this transform:

```suggestion
```